redisinsight/api/src/modules/redis/client/node-redis/cluster.node-redis.client.ts

```diff
@@ -95,4 +95,45 @@ export class ClusterNodeRedisClient extends NodeRedisClient {
   async call(command: RedisClientCommand, options?: IRedisClientCommandOptions): Promise<RedisClientCommandReply> {
     return this.sendCommand(command, options);
   }
+
+  /**
+   * @inheritDoc
+   */
+  async publish(channel: string, message: string): Promise<number> {
+    return this.client.publish(channel, message);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async subscribe(channel: string): Promise<void> {
+    const listener = (message: string, messageChannel : string) => {
+      this.emit('message', messageChannel, message);
+    };
+    await this.client.subscribe(channel, listener);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async pSubscribe(channel: string): Promise<void> {
+    const listener = (message: string, messageChannel : string) => {
+      this.emit('pmessage', channel, messageChannel, message);
+    };
+    await this.client.pSubscribe(channel, listener);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async unsubscribe(channel: string): Promise<void> {
+    await this.client.unsubscribe(channel);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  async pUnsubscribe(channel: string): Promise<void> {
+    await this.client.pUnsubscribe(channel);
+  }
```

Comment:
Can we put this logic to abstract ioredis.client class to not duplicate it for standalone, cluster and sentinel?